cl/assets/react/_disclosure_helpers.ts

```diff
@@ -0,0 +1,47 @@
+import {GROSS_VALUE, INCOME_GAIN, VALUATION_METHODS} from "./_disclosure_models";
+
+export const convertTD = (value, table, key) => {
+  if (value == -1) {
+    return ""
+  }
+  if (table == "Investments" && key == "transaction_value_code" && value) {
+    return `${GROSS_VALUE[value]} (${value})`
+  }
+  if (table == "Debts" && key == "value_code" && value) {
+    return `${GROSS_VALUE[value]} (${value})`
+  }
+  if (table == "Investments" && key == "income_during_reporting_period_code" && value) {
+    return `${INCOME_GAIN[value]} (${value})`
+  }
+  if (table == "Investments" && key == "gross_value_method" && value) {
+    return `${VALUATION_METHODS[value]} (${value})`
+  }
+  if (table == "Investments" && key == "gross_value_code" && value) {
+    return `${GROSS_VALUE[value]} (${value})`
+  }
+  if (table == "Investments" && key == "transaction_gain_code" && value) {
+    return `${INCOME_GAIN[value]} (${value})`
+  }
```

Comment:
Can we lump together the blocks on line 8, 11, and 20 at least?

Reply:
you got me just in time.  I was about to show you that I had the homepage working but should be an easy port over to the sidebar.... I think.

Reply:
this has been made much prettier

Reply:
and faster.